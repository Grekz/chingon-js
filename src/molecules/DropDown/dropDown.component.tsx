import styled from '@emotion/styled'
import * as React from 'react'

import Button from '../../atoms/Button'
import { ButtonTypes } from '../../atoms/Button/button.component'
import ListItem from '../../atoms/ListItem'
import List from '../List'

interface IDropDownProps {
  options?: string[]
  title?: string
}

type DropDownType = React.SFC<IDropDownProps>

const DropDown = ({ title = '', options }: IDropDownProps) => {
  const [isOpen, setOpen] = React.useState(false)
  const handleOnClick = () => setOpen(!isOpen)
  return (
    <DropDownWrapper>
      {options && title ? (
        <>
          <Button buttonType={ButtonTypes.PRIMARY} onClick={handleOnClick}>
            {title}
          </Button>
          {isOpen ? (
            <List>
              {options.map((item, index) => (
                <ListItem key={index}>{item}</ListItem>
              ))}
            </List>
          ) : null}
        </>
      ) : null}
    </DropDownWrapper>
  )
}

const DropDownWrapper: DropDownType = styled.div`
  position: relative;
  > button::after {
    display: inline-block;
    margin-left: 0.255em;
    vertical-align: 0.255em;
    content: '';
    border-top: 0.3em solid;
    border-right: 0.3em solid transparent;
    border-bottom: 0;
    border-left: 0.3em solid transparent;
    margin-left: 5px;
  }
`
export default DropDown
