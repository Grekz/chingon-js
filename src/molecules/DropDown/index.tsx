export { default } from './dropDown.component'
