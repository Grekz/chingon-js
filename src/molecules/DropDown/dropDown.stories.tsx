import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import DropDown from '.'

storiesOf('Molecules|DropDown', module).add('As Default', () => (
  <DropDown
    title="My drop down"
    options={['Option 1', 'Option 2', 'Option 3']}
  />
))
