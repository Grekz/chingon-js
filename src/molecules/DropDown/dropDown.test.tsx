import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import DropDown from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<DropDown /> rendering', () => {
  it('should render DropDown', () => {
    const wrapper = shallow(
      <DropDown
        title="My drop down"
        options={['Option 1', 'Option 2', 'Option 3']}
      />
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <DropDown
        title="My drop down"
        options={['Option 1', 'Option 2', 'Option 3']}
      />
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
