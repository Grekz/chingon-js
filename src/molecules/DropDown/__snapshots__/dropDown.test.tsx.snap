// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDown /> rendering should match snapshot 1`] = `
<Styled(div)>
  <Button
    buttonType="primary"
    onClick={[Function]}
  >
    My drop down
  </Button>
</Styled(div)>
`;
